### PR TITLE
[WFLY-11031] Introduce a console-access-log for web access logging

### PR DESCRIPTION
**Pull request by jamezp**
https://issues.jboss.org/browse/WFLY-11031

**Note:** This PR currently requires a SNAPSHOT build of WildFly Core until Beta5 is release.

@fl4via Can you please have a look at this as I went a slight different route than what we discussed previously. I added a `setting=console-access-log` resource instead of keeping everything under the `setting=access-log` resource. I felt it made the resource too complicated.

There had been some email discussions with using the standard `pattern` attribute for determining the exchange attributes to include in the JSON structured output. While I realize these are well known patterns I felt it just didn't fit right when we need the ability for defined keys in structured output. I went with a complex `attributes` management model attribute with known exchange attribute types.

### CODE REVIEW

**Comment by jmesnil** [comment]
@jamezp #12266 contains the Core 9.0.0.Beta5 upgrade that is required by this PR.

File: undertow/src/main/resources/org/wildfly/extension/undertow/LocalDescriptions.properties
Comment (jstourac) [inline undertow/src/main/resources/org/wildfly/extension/undertow/LocalDescriptions.properties]: Is this really used? I don't see it as an attribute in `access-log` element.

Comment (jamezp) [reply undertow/src/main/resources/org/wildfly/extension/undertow/LocalDescriptions.properties]: Nope, good catch. I'll get that removed thanks!

**Review by jstourac — APPROVED** [review]

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Undertow.adoc
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Undertow.adoc]: There should also be a paragraph like...

The `metadata` management model attribute can be used to add extra metadata to each log record. The value of the attribute is an arbitrary set of key/value pairs. The values can include management model expressions, which must be resolvable when the console access log service is started. The value is resolved once per start or reload of the server.

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/access/log/ConsoleAccessLogTestCase.java
Comment (fl4via) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/access/log/ConsoleAccessLogTestCase.java]: @jamezp I'm okay with having this, but lets keep an eye to see if we don't have random failures on this test when run on slower machines. If that happens, we will have to switch to an assertion in loop: if fail to find expected lines-> try to find lines again, unless timeout of, say, 30 s or 60 s has passed.

**Review by fl4via — APPROVED** [review]

**Review by fl4via — APPROVED** [review]

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/access/log/ConsoleAccessLogTestCase.java
Comment (jamezp) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/access/log/ConsoleAccessLogTestCase.java]: +1. I do not generally like things like this, but there seemed to be no other way :)

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Undertow.adoc
Comment (bstansberry) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Undertow.adoc]: I'm not going to block merging for this as I want it in but the value is a *set* of arbitrary key/value pairs, not a single pair. The CLI example is a set.

Comment (jamezp) [reply docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Undertow.adoc]: Good catch. I went ahead and fixed that.

**Comment by jamezp** [comment]
It looks like I should not have pushed the last change I did. I didn't notice #12300 and only looked at master-ignore by mistake. This one may need to be manually close and I apologize for that. Once #12300 is merged I'll make a new PR for the text update on the documentation.

**Comment by bstansberry** [comment]
It's fine; I'll merge this. I'm not merging #12300; it's just for CI.  But the only diff here vs what I tested is in the asciidoc.

**Pull request merged**
